api/envoy/api/v2/route/route.proto

```diff
@@ -1051,6 +1051,30 @@ message Decorator {
 }
 
 message Tracing {
+  // Describes custom tags for the active span.
+  message CustomTag {
+    // Type of custom tags.
+    enum Type {
+      LITERAL = 0;
```

Comment:
Please add documentation for each item. Also, we historically have had issues with enums like this as they are hard to extend and manage. Can you refactor so that each type is a message, and then there is a oneof that selects the type?

Reply:
fixed